crates/sui-indexer-alt/src/pipeline/concurrent/watermark.rs

```diff
@@ -132,15 +120,22 @@ pub(super) fn watermark<H: Handler + 'static>(
                             break;
                         }
 
-                        match watermark.next_cmp(&part.watermark) {
-                            Ordering::Future => break,
+                        match next_checkpoint.cmp(&part.watermark.checkpoint_hi_inclusive) {
+                            // Next pending checkpoint is from the future.
+                            Ordering::Less => break,
```

Comment:
doesnt this mean `next_checkpoint` is less than `watermark.checkpoint_hi_inclusive` which should instead mean that the pending checkpoint is from the past?

Reply:
nvm, we're comparing `next_checkpoint` against what has been indexed